tests/Database/DatabaseConnectionTest.php

```diff
@@ -148,14 +147,12 @@ public function testTransactionLevelNotIncrementedOnTransactionException()
         }
     }
 
-    public function testBeginTransactionMethodRetriesOnFailure()
+    public function testBeginTransactionMethodUsesCreateTransactionMethod()
     {
         $pdo = $this->createMock(DatabaseConnectionTestMockPDO::class);
-        $pdo->expects($this->at(0))
-            ->method('beginTransaction')
-            ->will($this->throwException(new ErrorException('server has gone away')));
-        $connection = $this->getMockConnection(['reconnect'], $pdo);
-        $connection->expects($this->once())->method('reconnect');
+        $connection = $this->getMockConnection(['createTransaction', 'fireConnectionEvent'], $pdo);
+        $connection->expects($this->once())->method('createTransaction');
+        $connection->expects($this->once())->method('fireConnectionEvent')->with('beganTransaction');
         $connection->beginTransaction();
         $this->assertEquals(1, $connection->transactionLevel());
```

Comment:
This test has become an entirely different one. I think it's best that we revert this part as well and try to tackle it together with those other tests?

Reply:
I have no idea how to fix this one without these changes! I'm removing `at` here, but original code is trying to throw exception in first call and do regular job in second call. Any idea?

Reply:
Let's just revert this for now and open a new PR later on where we can discuss this 👍

Reply:
:thinking: but whole purpose of PR will be failed. 
OK, i will revert it and i will change it with those two tests to cover what should be covered in first place.